### DISCUSSION TITLE
Difference between VM Disk CMK encryption for OS and Data disks

### DISCUSSION THREAD

**Discussion by tyconsulting** [Azure/ResourceModules#2235] [Ideas]
https://github.com/Azure/ResourceModules/blob/317cddea046b0858f12b743edbc4922d31bbebb2/modules/Microsoft.Compute/virtualMachines/deploy.bicep#L409

for the VM disk CMK in CARML module, I think the logic is not ideal comparing os disk vs data disk. if I specified enable encryption, why treating os disks and data disks differently? if I created a diskEncryptionSet, shouldn't it be used for both data and OS disks? currently, for OS disk, you will need to specify the resource Id for an existing diskEncryptionSet, but for Data disk, it will just create one for you. why is that?

**Comment by eriqua** [comment]
Hi @tyconsulting, thanks for your question.

I created issue #2237 from this discussion to provide more visibility to the request and allow its triage.
